### ISSUE TITLE
App freezes when stopping recording higher resolutions/framerates with GLESTextureInput

### ISSUE THREAD

**Issue by emielch** [natmlx/NatCorder#4]
I'm using NatCorder 1.8.6 to record a camera in my app. With _UseGLESTextureInput_ set to true the app often freezes when stopping the recording. This seems to happen more often as resolution and/or framerate increase. Logcat doesn't seem to show anything useful since the app just hangs.
I built my app twice, once limiting the framerate to 60 fps and once to 120 fps by setting _Application.targetFrameRate_ appropriately, and kept track of how often it freezes at various recording resolutions. Here are the results:

**60 fps**
- 1024x512&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0/5 failed
- 1536x768&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0/5 failed
- 2048x1024&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0/5 failed
- 2560x1280&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2/5 failed
- 3072x1536&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5/5 failed

**120 fps**

- 1024x512&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0/5 failed
- 1536x768&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3/5 failed
- 2048x1024&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3/5 failed
- 2560x1280&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5/5 failed
- 3072x1536&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5/5 failed

This is on Android (Sony Xperia 5 ii), and Unity 2021.3.7. Also tested on a Samsung Galaxy S7

**Comment by olokobayusuf** [comment]
Reproduced with the highest resolution.

**Comment by olokobayusuf** [comment]
Fixed! Coming in 1.8.7.

**Comment by emielch** [comment]
Awesome! Looking forward to the release

**Comment by olokobayusuf** [comment]
Fixed in [1.8.7](https://github.com/natmlx/NatCorder/releases/tag/1.8.7).

**Comment by emielch** [comment]
I updated to 1.8.7 and the problem has been reduced but is not gone yet.
At 60 fps, 2560x1280 now finishes successfully 100% of the time, and at 3072x1536 mostly as well (it froze once).
At 120 fps I still got freezes sometimes at 2048x1024 (success rate seems to improve if I record at a lower resolution beforehand) and it always freezes at 2560x1280.

**Comment by olokobayusuf** [comment]
Reopening for further investigation.

**Comment by olokobayusuf** [comment]
@emielch I can't reproduce this on my S21. How are you able to run the app at 120FPS? I don't think Unity supports rendering at 120FPS on Android. Also, any reason why you are recording at such high resolutions?

**Comment by emielch** [comment]
@olokobayusuf Thanks for looking into it. Unity definitely supports it. You'll have to set the target framerate to 120 though (I think by default Unity sets it to 30 for mobile devices). I just put this somewhere in the code:
```
private void Awake() {
   Application.targetFrameRate = 120;
}
```
My phone (Xperia 5 II) has a "Game enhancer" that by default limits games to 60 fps to save power. I could set it to 120 fps by enabling a custom profile for my game. I don't know if the S21 requires something similar.

To the question why, simply because it (used to) can. The app is about interactively generating animated art with a lot of small scale details that look amazing at high resolutions / framerates. I simply record the canvas at the same resolution as it gets generated, so what you see is what you get (or what you get is even more if you play it back on a larger screen afterwards) and it looks amazing. This used to work well with NatCorder 1.3. When rendering at high resolutions @ 120fps the framerate would drop while recording if the device couldn't keep up, but it could still be higher than 60 and I liked the fact that it "automatically" adjusts its framerate so it makes maximum use of the device's capabilities.

**Comment by olokobayusuf** [comment]
Fixed in 1.8.8.
